beacon_node/beacon_chain/tests/store_tests.rs

```diff
@@ -512,14 +512,16 @@ async fn epoch_boundary_state_attestation_processing() {
             .unwrap()
             .expect("block exists");
         let mut epoch_boundary_state = store
-            .load_epoch_boundary_state(&block.state_root())
+            .load_hot_state(&block.state_root())
             .expect("no error")
-            .expect("epoch boundary state exists");
+            .expect("epoch boundary state exists")
+            .0;
         let ebs_state_root = epoch_boundary_state.update_tree_hash_cache().unwrap();
         let mut ebs_of_ebs = store
```

Comment:
@michaelsproul I'm confused by this test. It appears to do:
- load state with `load_epoch_boundary_state` for root A
- compute the state root of above state, which _should_ be equal to root A
- load state with `load_epoch_boundary_state` for root A (again)

If that's correct why does it load the state twice? The test is failing because some attestations are later than finalized and `load_hot_state` returns None as the state is in the cold DB

Reply:
Ah `load_epoch_boundary_state` in unstable looks in both DBs. First hot db, then if None defaults to cold. So switching to get_state is the correct fix. I applied it in https://github.com/sigp/lighthouse/pull/7176/commits/2b050f4e8992a38af0fef116d0dde30f97241b9e